### PR TITLE
Hvg fix pcs

### PR DESCRIPTION
**Pull request by swalpe**
Modifs to the HVG function to accommodate for batch-aware analyses. It seems to me that there was a double-log in the reculter function & I believe to have removed that, can u pls double-check that all looks k?

### CODE REVIEW

**Review by hatjek — APPROVED** [review]
Looks good to me.

File: besca/st/_wrapper_funcs.py [hunk 1]
```diff
@@ -261,18 +261,23 @@ def clr_normalize(adata, results_folder):
     return(adata)
 
 
-def highly_variable_genes(adata, batch_key=None):
+def highly_variable_genes(adata, batch_key=None, n_shared=2):
```
Comment (ajulienla) [inline besca/st/_wrapper_funcs.py:264]: I think this should be quickly documented. It was never done.Could be the good occasion

File: besca/tl/rc/_reclustering.py [hunk 2]
```diff
@@ -90,21 +93,27 @@ def recluster(adata,
...
+        pl_highly_variable_genes(cluster_subset, show=True)
+    print('In total', str(sum(cluster_subset.var.highly_variable)), 'highly variable genes selected within cluster')
 
     #apply filter
-    cluster_subset = _subset_adata(cluster_subset, filter_result.highly_variable, axis = 1, raw = False)
+    cluster_subset = _subset_adata(cluster_subset, cluster_subset.var.highly_variable, axis = 1, raw = False)
 
     #perform further processing
-    log1p(cluster_subset)
+    # log1p(cluster_subset) # data already logged
```
Comment (ajulienla) [inline besca/tl/rc/_reclustering.py:113]: We could remove the line. It would be clearer ?

**Review by ajulienla — APPROVED** [review]
Great. Thanks for the update. 
I think it could be a good occasion to document the highly variable gene wrapper function.

File: besca/st/_wrapper_funcs.py (hunk 1 above)
Comment (swalpe) [reply besca/st/_wrapper_funcs.py:264]: Ok done

File: besca/tl/rc/_reclustering.py (hunk 2 above)
Comment (swalpe) [reply besca/tl/rc/_reclustering.py:113]: I would keep it, just if in case we missed something here

**Pull request merged**